src/wmtk/operations/tri_mesh/EdgeSwapValence.cpp

```diff
@@ -0,0 +1,61 @@
+#include "EdgeSwapValence.hpp"
+#include <wmtk/SimplicialComplex.hpp>
+#include <wmtk/TriMesh.hpp>
+#include "EdgeCollapse.hpp"
+#include "EdgeSplit.hpp"
+namespace wmtk::operations::tri_mesh {
+EdgeSwapValence::EdgeSwapValence(
+    Mesh& m,
+    const Tuple& t,
+    const OperationSettings<EdgeSwapValence>& settings)
+    : EdgeSwapBase(m, t, settings.base_settings)
+    , m_settings{settings}
+{}
+
+std::string EdgeSwapValence::name() const
+{
+    return "tri_mesh_edge_swap_valence";
+}
+
+bool EdgeSwapValence::execute()
+{
+    const Tuple v0 = input_tuple();
```

Comment:
could we implement this valence thing in the standard swap? we just have a bool for if we want to do it / have a std::optional<long> initial_valence` ?. after checkpoints/if checkpoints gets finished this should just be an invariant anyway